Analyzer/src/TwoLepAnalyzer.cc

```diff
@@ -0,0 +1,442 @@
+            {
+                lep1 = GoodLeptons.at(1).second;
+                lep2 = GoodLeptons.at(0).second;
+            }
+            if(NGoodBJets_pt30 >= 2)
+            {
+                for(int b=0; b < NGoodBJets_pt30; b++)
+                {
+                    double  mbl1 = (lep1 + GoodBJetsVec_pt30.at(b)).M();
+                    if( abs(mbl1 - 105) < mbl1diff)
+                    {
+                        Mbl1 = mbl1;
+                        mbl1diff = abs(mbl1 - 105);
+                        used_jet = b;
+                    }
+                }
+                for(int b=0; b < NGoodBJets_pt30; b++)
+                {
+                    if(b != used_jet)
+                    {
+                        double mbl2 = (lep2 + GoodBJetsVec_pt30.at(b)).M();
+                        if( abs(mbl2 - 105) < mbl2diff)
+                        {
+                            Mbl2 = mbl2;
+                            mbl2diff = abs(mbl2 - 105);
+                        }
+                    }
+                }
+            }
+            else if (NGoodBJets_pt30 == 1)
+            {
+                if ( abs((lep1 + GoodBJetsVec_pt30.at(0)).M() - 110) <= abs((lep2 + GoodBJetsVec_pt30.at(0)).M() - 110) )
+                {
+                    Mbl1 = (lep1 + GoodBJetsVec_pt30.at(0)).M();
+                    for (int j = 0; j < Jets.size(); j++)
+                    {
+                        double mbl2 = (lep2 + Jets.at(j)).M();
+                        if (abs(mbl2 - 105) < mbl2diff && !GoodBJets_pt30.at(j))
+                        {
+                            Mbl2 = mbl2;
+                            mbl2diff = abs(mbl2 - 105);
+                        }
+                    }
+                }
+                else
+                { 
+                    Mbl2 = (lep2 + GoodBJetsVec_pt30.at(0)).M();
+                    for (int j = 0; j < Jets.size(); j++)
+                    {
+                        double mbl1 = (lep1 + Jets.at(j)).M();
+                        if (abs(mbl1 - 105) < mbl1diff && !GoodBJets_pt30.at(j))
+                        {
+                            Mbl1 = mbl1;
+                            mbl1diff = abs(mbl1 - 105);
+                        }
+                    }
+                }
+            }
+            else            
+            {
+                for(int j=0; j < Jets.size(); j++)
+                {
+                    double  mbl1 = (lep1 + Jets.at(j)).M();
+                    if( abs(mbl1 - 105) < mbl1diff)
+                    {
+                        Mbl1 = mbl1;
+                        mbl1diff = abs(mbl1 - 105);
+                        used_jet = j;
+                    }
+                }
+                for(int j=0; j < Jets.size(); j++)
+                {
+                    if( j != used_jet)
+                    {                    
+                        double mbl2 = (lep2 + Jets.at(j)).M();
+                        if( abs(mbl2 - 105) < mbl2diff)
```

Comment:
You have been using this if statement over and over again. This means you should make a function that does this.